core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -838,11 +838,11 @@ object TestUtils extends Logging {
    * Wait until the given condition is true or throw an exception if the given wait time elapses.
    */
   def waitUntilTrue(condition: () => Boolean, msg: => String,
-                    waitTime: Long = JTestUtils.DEFAULT_MAX_WAIT_MS, pause: Long = 100L): Boolean = {
+                    waitTime: Long = JTestUtils.DEFAULT_MAX_WAIT_MS, pause: Long = 100L): Unit = {
```

Comment:
The `procedural` style was preferred, but it's deprecated in Scala 2.12 and will be removed in a future version. That's why we have moved to this style.